File: Naheulbook.Web/Responses/MonsterTemplateResponse.cs
using System.Collections.Generic;
using Newtonsoft.Json.Linq;

// ReSharper disable UnusedMember.Global
// ReSharper disable AutoPropertyCanBeMadeGetOnly.Global

namespace Naheulbook.Web.Responses
{
    public class MonsterTemplateResponse
    {
        public class MonsterTemplateInventoryElementResponse
        {
            public int Id;
            public ItemTemplateResponse ItemTemplate = null!;
            public int MinCount;
            public int MaxCount;
            public float Chance;
        }

        public int Id { get; set; }
        public string Name { get; set; } = null!;
        public int SubCategoryId { get; set; }
        public JObject Data { get; set; } = null!;
        public List<MonsterTemplateInventoryElementResponse> SimpleInventory { get; set; } = null!;
    }
}

File: src/app/api/responses/monster-template-response.ts
import {ItemTemplateResponse} from './item-template-response';
import {MonsterTemplateData} from '../shared';

export interface MonsterTemplateResponse {
    id: number;
    name: string;
    subCategoryId: number;
    data: MonsterTemplateData;
    simpleInventory: MonsterSimpleInventoryResponse[];
}

export interface MonsterSimpleInventoryResponse {
    id: number;
    itemTemplate: ItemTemplateResponse;
    minCount: number;
    maxCount: number;
    chance: number;
    hidden: boolean;
}
